bciers/apps/reporting/src/app/bceidbusiness/industry_user_admin/reports/[version_id]/additional-reporting-data/page.tsx

```diff
@@ -1,15 +1,26 @@
 import { Suspense } from "react";
 import Loading from "@bciers/components/loading/SkeletonForm";
 import AdditionalReportingData from "@reporting/src/app/components/additionalInformation/AdditionalReportingData";
+import { getRegistrationPurpose } from "@reporting/src/app/utils/getRegistrationPurpose";
 
-export default async function Page({
-  params,
-}: {
+interface PageProps {
   params: { version_id: number };
-}) {
+}
+
+export default async function Page({ params }: PageProps) {
+  const versionId = params.version_id;
+  const registrationPurposes =
+    (await getRegistrationPurpose(versionId))?.registration_purposes || [];
+  const includeElectricityGenerated = registrationPurposes.includes(
+    "OBPS Regulated Operation",
+  );
+
   return (
     <Suspense fallback={<Loading />}>
-      <AdditionalReportingData versionId={params.version_id} />
+      <AdditionalReportingData
+        versionId={versionId}
+        includeElectricityGenerated={includeElectricityGenerated}
+      />
     </Suspense>
   );
```

Comment:
This code is now duplicated between the multiple page in the different roles.
We should put that in an `AdditionalReportingData` server-side component instead, next to the other one

We've been (loosely) following this pattern in other places:
`AdditionalReportingData.tsx` : server-side component with initial data fetching/computing logic
`AdditionalReportingDataForm.tsx`: client-side component with the form and the interaction